libraries/service/src/main/java/org/holochain/androidserviceruntime/service/HolochainService.kt

```diff
@@ -30,22 +39,53 @@ class HolochainService : Service() {
     private val serviceScope = CoroutineScope(supervisorJob)
     private val adminBinder by lazy { AdminBinder() }
     private val appBinders = mutableMapOf<String, AppBinder>()
+    private var appAuthorizationRequests = AppAuthorizationRequestMap(NOTIFICATION_ID_FOREGROUND_SERVICE_RUNNING + 1)
+
+    companion object {
+        // Reserved notification id for service running notification
+        const val NOTIFICATION_ID_FOREGROUND_SERVICE_RUNNING = 1
+
+        // Notification Channel Id for service running notification
+        const val NOTIFICATION_CHANNEL_ID_FOREGROUND_SERVICE = "org.holochain.androidserviceruntime.service.FOREGROUND_SERVICE"
+
+        // Notification Channel Id for app authorization notifications
+        const val NOTIFICATION_CHANNEL_ID_APP_AUTHORIZATION = "org.holochain.androidserviceruntime.service.APP_AUTHORIZATION"
+
+        // Approve app authorization request
+        const val ACTION_APPROVE_APP_AUTHORIZATION = "org.holochain.androidserviceruntime.service.APPROVE_APP_AUTHORIZATION"
+
+        // Deny app authorization request
+        const val ACTION_DENY_APP_AUTHORIZATION = "org.holochain.androidserviceruntime.service.DENY_APP_AUTHORIZATION"
+
+        // Start the Service, and the Holochain conductor
+        const val ACTION_START = "org.holochain.androidserviceruntime.service.ACTION_START"
+    }
 
     private inner class AdminBinder : IHolochainServiceAdmin.Stub() {
         private val logTag = "IHolochainServiceAdmin"
```

Comment:
Changes below this comment are plumbing the service's Admin and App ipc interfaces to actually callback with an exception if the client is not authorized.